lib/ethers/transaction/eip4844.ex

```diff
@@ -0,0 +1,170 @@
+  - `chain_id` - chain ID of network where the transaction is to be executed
+  - `nonce` - sequence number for the transaction from this sender
+  - `max_priority_fee_per_gas` - maximum fee per gas (in wei) to give to validators as priority fee (introduced in EIP-1559)
+  - `max_fee_per_gas` - maximum total fee per gas (in wei) willing to pay (introduced in EIP-1559)
+  - `gas` - maximum amount of gas allowed for transaction execution
+  - `to` - destination address for transaction, nil for contract creation
+  - `value` - amount of ether (in wei) to transfer
+  - `input` - data payload of the transaction
+  - `access_list` - list of addresses and storage keys to warm up (introduced in EIP-2930)
+  - `max_fee_per_blob_gas` - maximum fee per blob gas (in wei) willing to pay (introduced in EIP-4844)
+  - `blob_versioned_hashes` - list of versioned hashes of the blobs (introduced in EIP-4844)
+  """
+  @type t :: %__MODULE__{
+          chain_id: non_neg_integer(),
+          nonce: non_neg_integer(),
+          max_priority_fee_per_gas: non_neg_integer(),
+          max_fee_per_gas: non_neg_integer(),
+          gas: non_neg_integer(),
+          to: Types.t_address() | nil,
+          value: non_neg_integer(),
+          input: binary(),
+          access_list: [{binary(), [binary()]}],
+          max_fee_per_blob_gas: non_neg_integer(),
+          blob_versioned_hashes: [{binary(), [binary()]}]
+        }
+
+  @impl Ethers.Transaction
+  def new(params) do
+    {:ok, eip1559_data} = Eip1559.new(params)
+
+    {:ok,
+     %__MODULE__{
+       chain_id: eip1559_data.chain_id,
+       nonce: eip1559_data.nonce,
+       max_priority_fee_per_gas: eip1559_data.max_priority_fee_per_gas,
+       max_fee_per_gas: eip1559_data.max_fee_per_gas,
+       gas: eip1559_data.gas,
+       to: eip1559_data.to,
+       value: eip1559_data.value,
+       input: eip1559_data.input,
+       access_list: eip1559_data.access_list,
+       max_fee_per_blob_gas: params.max_fee_per_blob_gas,
+       blob_versioned_hashes: params[:blob_versioned_hashes] || []
+     }}
+  end
+
+  @impl Ethers.Transaction
+  def auto_fetchable_fields do
+    [:max_fee_per_blob_gas | Eip1559.auto_fetchable_fields()]
+  end
+
+  @impl Ethers.Transaction
+  def type_envelope, do: <<type_id()>>
+
+  @impl Ethers.Transaction
+  def type_id, do: @type_id
+
+  @impl Ethers.Transaction
+  def from_rlp_list([
+        chain_id,
+        nonce,
+        max_priority_fee_per_gas,
+        max_fee_per_gas,
+        gas,
+        to,
+        value,
+        input,
+        access_list,
+        max_fee_per_blob_gas,
+        blob_versioned_hashes
+        | rest
+      ]) do
+    case Eip1559.from_rlp_list([
+           chain_id,
+           nonce,
+           max_priority_fee_per_gas,
+           max_fee_per_gas,
+           gas,
+           to,
+           value,
+           input,
+           access_list
+         ]) do
```

Comment:
I actually think copy/pasting the logic from Eip1559 would make it more readable and easier to maintain. WDYT?

Reply:
I did think about this before, doing what you suggest would be more readable (less referencing) but imo is harder to maintain. But probably not that much harder as EIP-1559 logic will not change in the future. tbh is a tricky situation copying logic vs reusing logic for readability

Reply:
About the maintenance part, as there are no expectations that either of these implementations will change, I think it will not matter that much. In this specific case I personally lean a tad bit on the readability side. But if you feel strong about it I'm okay with calling Eip1559 module too.

Reply:
Flipped a coin, will change to suit readability 👍